shared/actions/chat/index.js

```diff
@@ -346,61 +346,94 @@ function * _loadMoreMessages (action: Constants.LoadMoreMessages): SagaGenerator
+    const messages = (thread && thread.messages || []).map(message => _unboxedToMessage(message, yourName, yourDeviceName, conversationIDKey)).reverse()
+    let newMessages = []
+    messages.forEach((message, idx) => {
+      if (idx > 0) {
+        const timestamp = Shared.maybeAddTimestamp(messages[idx], messages[idx - 1])
+        if (timestamp !== null) {
+          newMessages.push(timestamp)
+        }
       }
-    }
-    newMessages.push(message)
-  })
+      newMessages.push(message)
+    })
 
-  const pagination = _threadToPagination(thread)
+    const pagination = _threadToPagination(thread)
 
-  yield put({
-    payload: {
-      conversationIDKey,
-      messages: newMessages,
-      moreToLoad: !pagination.last,
-      paginationNext: pagination.next,
+    yield put({
+      logTransformer: Shared.prependMessagesActionTransformer,
+      payload: {
+        conversationIDKey,
+        messages: newMessages,
+        moreToLoad: !pagination.last,
+        paginationNext: pagination.next,
+      },
+      type: 'chat:prependMessages',
+    })
+
+    // Load previews for attachments
+    const attachmentsOnly = messages.reduce((acc: List<Constants.AttachmentMessage>, m) => m && m.type === 'Attachment' && m.messageID ? acc.push(m) : acc, new List())
+    // $FlowIssue we check for messageID existance above
+    yield attachmentsOnly.map(({conversationIDKey, messageID, filename}: Constants.AttachmentMessage) => put(Creators.loadAttachment(conversationIDKey, messageID, true, false, tmpFile(Shared.tmpFileName(false, conversationIDKey, messageID, filename))))).toArray()
+  }
+
+  const channelConfig = singleFixedChannelConfig([
+    'chat.1.chatUi.chatThreadCached',
+    'chat.1.chatUi.chatThreadFull',
+    'finished',
+  ])
+
+  const loadInboxChanMap: ChannelMap<any> = ChatTypes.localGetThreadNonblockRpcChannelMap(channelConfig, {
+    param: {
+      conversationID,
+      identifyBehavior: TlfKeysTLFIdentifyBehavior.chatGui,
+      pagination: {
+        last: false,
+        next,
+        num: Constants.maxMessagesToLoadAtATime,
+        previous: null,
+      },
+      query: {
+        disableResolveSupersedes: false,
+        markAsRead: true,
+        messageTypes,
+      },
     },
-    logTransformer: Shared.prependMessagesActionTransformer,
-    type: 'chat:prependMessages',
   })
 
-  // Load previews for attachments
-  const attachmentsOnly = messages.reduce((acc: List<Constants.AttachmentMessage>, m) => m && m.type === 'Attachment' && m.messageID ? acc.push(m) : acc, new List())
-  // $FlowIssue we check for messageID existance above
-  yield attachmentsOnly.map(({conversationIDKey, messageID, filename}: Constants.AttachmentMessage) => put(Creators.loadAttachment(conversationIDKey, messageID, true, false, tmpFile(Shared.tmpFileName(false, conversationIDKey, messageID, filename))))).toArray()
+  while (true) {
+    const incoming: {[key: string]: any} = yield race({
```

Comment:
can we have two of these for the same conversation? since we are doing take every vs take latest. Or does the cancel/clearThread handle that case?